common/network-yarn/src/main/java/org/apache/spark/network/yarn/YarnShuffleService.java

```diff
@@ -297,6 +306,7 @@ public void stopApplication(ApplicationTerminationContext context) {
         secretManager.unregisterApp(appId);
       }
       blockHandler.applicationRemoved(appId, false /* clean up local dirs */);
+      shuffleMergeManager.applicationRemoved(appId, false);
```

Comment:
This should mirror what we do for `initializeApplication` ?

Reply:
This is removed from here and moved to `ExternalBlockHandler.applicationRemoved`